Mathlib/Tactic/Linter/MinImports.lean

```diff
@@ -53,18 +67,29 @@ namespace MinImports
 
 open Mathlib.Command.MinImports
 
+/-- `impsBelow env ms` takes as input an `Environment` `env` and a `NameSet` of module names `ms`.
+It returns the modules that are transitively imported by `ms`.
+-/
+def importsBelow (ig : NameMap (Array Name)) (ms : NameSet) : NameSet :=
+  (ig.upstreamOf ms).fold (σ := NameSet) (fun _ ↦ ·.insert ·) {}
```

Comment:
We could avoid recalculating here, too. Rather than stashing `env.importGraph` in the state, stash `(← env.importGraph).transitiveClosure`.

Reply:
I'm not sure that I understand here: `importGraph` is a `NameMap (Array Name)`, while `transitiveClosure` returns a `NameMap NameSet`.  I do not really know how to then use `upstreamOf`, since that function wants an `importGraph` as an input.

Reply:
You wouldn't even need upstreamOf. You'd just take the union over `ms` of the values from the `transitiveClosure` map.

Reply:
Ah, I understand now, thanks!  I updated the PR.